File: src/main/java/vocadb/notification/reader/client/query/FollowedArtistSortRule.java
package vocadb.notification.reader.client.query;

import com.fasterxml.jackson.annotation.JsonProperty;

public enum FollowedArtistSortRule {
    @JsonProperty("None")
    NONE,
    @JsonProperty("AdditionDate")
    ADDITION_DATE,
    @JsonProperty("AdditionDateAsc")
    ADDITION_DATE_ASC,
    @JsonProperty("FollowerCount")
    FOLLOWER_COUNT,
    @JsonProperty("Name")
    NAME,
    @JsonProperty("ReleaseDate")
    RELEASE_DATE,
    @JsonProperty("SongCount")
    SONG_COUNT,
    @JsonProperty("SongRating")
    SONG_RATING;
}

File: src/main/java/vocadb/notification/reader/client/api/enums/FollowedArtistSortRule.kt
package vocadb.notification.reader.client.api.enums

import com.fasterxml.jackson.annotation.JsonValue

enum class FollowedArtistSortRule(@JsonValue val value: String) {
    NONE("None"),

    ADDITION_DATE("AdditionDate"),
    ADDITION_DATE_ASC("AdditionDateAsc"),
    FOLLOWER_COUNT("FollowerCount"),
    NAME("Name"),
    RELEASE_DATE("ReleaseDate"),
    SONG_COUNT("SongCount"),
    SONG_RATING("SongRating");
}
